### DISCUSSION TITLE
Delta Neutral Portfolio with constraints on the underlyings

### DISCUSSION THREAD

**Discussion by alen-pavlovic** [dcajasn/Riskfolio-Lib#145] [Q&A]
hello everyone,

I am trying to create a delta-neutral portfolio with 4 different derivatives. 
.e.g

BTC swap Binance
BTC swap OKEx
ETH swap Binance
ETH swap OKEx

I want my portfolio to be delta neutral on the INSTRUMENT level. What I mean is to have BTC delta = 0 and ETH delta = 0.

Is it possible via riskfolio?

**Comment by alen-pavlovic** [comment]
anyone?

**Comment by alen-pavlovic** [comment]
asset_classes = {
                    'Assets': [
                                 'BTCBUSD_SWP_BN', 'BTCUSDT_SWP_BN', 'BTCUSDT_SWP_BY', 'BTCUSDT_SWP_GIO', 'BTCUSDT_SWP_HB', 'BTCUSDT_SWP_KC', 'BTCUSDT_SWP_OK', 'BTCUSD_SWP_DYDX',
                                 'ETHBUSD_SWP_BN', 'ETHUSDT_SWP_BN', 'ETHUSDT_SWP_BY', 'ETHUSDT_SWP_GIO', 'ETHUSDT_SWP_HB', 'ETHUSDT_SWP_KC', 'ETHUSDT_SWP_OK', 'ETHUSD_SWP_DYDX'
                               ], 
                    'Exchange1': [
                                  'BTC' , 'BTC', 'BTC', 'BTC' , 'BTC' , 'BTC' , 'BTC', 'BTC',
                                  'ETH' , 'ETH', 'ETH', 'ETH' , 'ETH' , 'ETH' , 'ETH', 'ETH'
                                  ],
                    'Exchange2': [
                                  'BTC' , 'BTC', 'BTC', 'BTC' , 'BTC' , 'BTC' , 'BTC', 'BTC',
                                  'ETH' , 'ETH', 'ETH', 'ETH' , 'ETH' , 'ETH' , 'ETH', 'ETH'    
                    ]
                 }

asset_classes = pd.DataFrame(asset_classes)
asset_classes = asset_classes.sort_values(by=['Assets'])

w = 0.001
constraints = {
              'Disabled': [False, False, False, False],
              'Type': ['Classes', 'Classes', 'Classes', 'Classes'],
              'Set': ['Exchange1', 'Exchange2', 'Exchange1', 'Exchange2'],
              'Position': ['BTC', 'BTC', 'ETH', 'ETH'],
              'Sign': ['<=', '>=', '<=', '>='],
              'Weight': [w, -w, w, -w],
              'Type Relative': ['', '', '', ''],
              'Relative Set': ['', '', '', ''],
              'Relative': ['', '', '', ''],
              'Factor': ['', '', '', '']
              }

constraints = pd.DataFrame(constraints)

display(constraints)

=======================================

I tried this and many other variations, doesn't work.

**Comment by alen-pavlovic** [comment]
@dcajasn any help?

**Reply by dcajasn** [reply]
Hi @allenpavlovich, I only fix errors in source code or add new features that seem interesting for me. For questions or help please read the readme file, consulting feed section.
